apps/account-api/src/services/delegation.service.ts

```diff
@@ -47,36 +46,31 @@ export class DelegationService {
     throw new Error('Invalid msaId.');
   }
 
-  async getRevokeDelegationPayload(accountId: string, providerId: string): Promise<RevokeDelegationPayloadResponseDto> {
-    if (!isAddress(accountId)) {
-      throw new HttpException('Invalid accountId', HttpStatus.BAD_REQUEST);
-    }
+  async getRevokeDelegationPayload(
+    accountId: string,
+    providerMsaId: string,
+  ): Promise<RevokeDelegationPayloadResponseDto> {
     const msaId = await this.blockchainService.publicKeyToMsaId(accountId);
     if (!msaId) {
       throw new HttpException('MSA ID for account not found', HttpStatus.NOT_FOUND);
     }
 
-    if (providerId) {
-      const isValidProviderId = await this.blockchainService.isValidMsaId(providerId);
```

Comment:
Validated via decorator semantically, but not on-chain. `BlockchainService.isValidMsaId(msaId)` validates that the given identifier is actually an MSA registered on-chain. (more specifically, validates that `msaId <= api.query.msa.currentMsaIdentifierMaximum()`, so technically it doesn't check if the MSA is retired, but still...)

Reply:
Ah I see what you are saying. I think it's a bit overkill to call the chain twice since we already doing it in `getProviderToRegistryEntry` but I don't have any strong opinions and will add that back